### DISCUSSION TITLE
TypedDict and missing keys

### DISCUSSION THREAD

**Discussion by dineshbvadhia** [python/typing#1883] [Q&A] [answered]
Hi. I don't understand why the following causes a `Missing keys` error?
```
class ConnectKwargs(TypedDict):
    host: str
    user: str
    port: str

class TransportParams(ConnectKwargs):
    connect_kwargs: ConnectKwargs
    keyfile: str | None
    passphrase: str | None
    
connect_kwargs: ConnectKwargs = {'host':payload['host'], 'user':payload['user'], 'port':payload['port']}
transport_params: TransportParams = {'connect_kwargs':connect_kwargs, 'keyfile':payload['keyfile'], 'passphrase':payload['passphrase']}
```    
`error: Missing keys ("host", "user", "port") for TypedDict "TransportParams"  [typeddict-item]
`

**Comment by brianschubert** [comment, marked as answer]
Since `TransportParams` is a subclass of `ConnectKwargs`, it inherits all of the keys of `ConnectKwargs`, more or less the same as if you had written
```python
class TransportParams(TypedDict):
    host: str
    user: str
    port: str
    connect_kwargs: ConnectKwargs
    keyfile: str | None
    passphrase: str | None
```

See [TypedDict - Inheritance](https://typing.readthedocs.io/en/latest/spec/typeddict.html#inheritance) from the typing spec for more details.
